### PR TITLE
IconButtonFloating: Add new component for floating actions

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

A new, simplified version of IconButton, that floats on content and is used to open a Dropdown or Modal
Screen Shot 2022-12-01 at 4 54 22 PM

#### Why?

Create consistency in the product for floating actions

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4999)
- [Figma](https://www.figma.com/file/uBMftqozsNhl0HJCs2e8Mo/Floating-IconButton?node-id=1%3A100&t=saTRhKTikKvp2j4r-0)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: packages/gestalt/src/IconButtonFloating.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      accessibilityHaspopup,
+      accessibilityLabel,
+      icon,
+      onClick,
+    } = props;
+
+    const innerRef = useRef(null);
+    // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
+    // that renders <IconButtonFloating ref={inputRef} /> to call inputRef.current.focus()
+    useImperativeHandle(ref, () => innerRef.current);
```
Comment (ayeshakmaz) [inline packages/gestalt/src/IconButtonFloating.js:47]: Not entirely sure why it doesn't like `ref` here

File: packages/gestalt/src/IconButtonFloating.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+// @flow strict
+import { forwardRef, type Node, useImperativeHandle, useRef } from 'react';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+import icons from './icons/index.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
```
Comment (dangerismycat) [inline packages/gestalt/src/IconButtonFloating.js:6]: We can't use this in any user-facing types as it is opaque to the user. The event type needs to be defined explicitly

File: docs/pages/web/iconbuttonfloating.js [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+import AccessibilitySection from '../../docs-components/AccessibilitySection.js';
+import GeneratedPropTable from '../../docs-components/GeneratedPropTable.js';
+
+export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen |}): Node {
+  return (
+    <Page title={generatedDocGen?.displayName}>
+      <PageHeader
+        name={generatedDocGen?.displayName}
+        description={generatedDocGen?.description}
+        defaultCode={`
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/pages/web/iconbuttonfloating.js:19]: I wouldn't use yet, as you wont be able to see the cmp.

[the remark this answers is not in the document]
Comment (ayeshakmaz) [reply docs/pages/web/iconbuttonfloating.js:19]: Yeah, can't use Sandpack yet

File: packages/gestalt/src/IconButtonFloating.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+type Props = {|
+  /**
+   * String that clients such as VoiceOver will read to describe the icon button. Always localize the label. See [Accessibility section](https://gestalt.pinterest.systems/web/modalalert#Accessibility) for more info.
+   */
+  accessibilityLabel: string,
+  icon?: $Keys<typeof icons>,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:13]: Is there a default? Should it be required?

I think it should be required. 
Screen Shot 2022-11-29 at 6 05 26 PM

File: packages/gestalt/src/IconButtonFloating.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  onClick?: AbstractEventHandler<
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:14]: yeah, look here to get an orientation packages/gestalt/src/SideNavigationTopItem.js

File: packages/gestalt/src/IconButtonFloating.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+/**
+ * A [IconButtonFloating](https://gestalt.pinterest.systems/web/iconbuttonfloating) represents the primary or most common action on the screen. As the name suggests, it floats over the content and is always on top of everything on the screen. Similar to the IconButton, the floating version uses icons instead of text to convey available actions; however, it is used when the action needs to be visible at all times in a sticky way where content can scroll underneath. IconButtonFloating remains in place on scroll.
+
+By default, it has a circular shape with a floating elevation shadow style built-in; when pressed, it will open more related actions by triggering a Dropdown or a Modal.
+
+IconButtonFloating is typically found in the Home feed, boards, and dashboards, supporting Pinners to perform core actions.
+ */
+const IconButtonFloatingWithForwardRef: React$AbstractComponent<Props, HTMLButtonElement> =
+  forwardRef<Props, HTMLButtonElement>(function IconButtonFloating(props: Props, ref): Node {
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:34]: cleanup

as a pattern, we destructure inside the function 

IconButtonFloating({ 
a,
b,
c
}: Props, ref)

File: packages/gestalt/src/IconButtonFloating.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      onClick,
+    } = props;
+
+    const innerRef = useRef(null);
+    // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
+    // that renders <IconButtonFloating ref={inputRef} /> to call inputRef.current.focus()
+    useImperativeHandle(ref, () => innerRef.current);
+
+    return (
+      <Box borderStyle="shadow" rounding="circle">
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:50]: The IconButton compresses but not the outer Box,that's why we observe a white halo.

Should we 1. 
wrap everything in a TapArea that compresses? >> More context here https://gestalt.pinterest.systems/web/taparea#link_buttons

2. Replace IconButton with Icon and wrap with TapArea?

Comment (ayeshakmaz) [reply packages/gestalt/src/IconButtonFloating.js]: Chatted w/ Cintia about this and we're leaving as is for now until we fix other color issues in Pog/Icon/IconButton

File: docs/pages/web/iconbuttonfloating.js [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+        `}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Best practices">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            description="Use when an action have to be visible at all times in a sticky way where content can scroll underneath."
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:112]: typo: "…when an action _has_ to be visible…"

File: docs/pages/web/iconbuttonfloating.js [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            description="Use IconButtonFloating for positive and supportive actions like Create, Help or Maximize."
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Use IconButtonFloating for negative and destructive actions like, Delete or Remove. "
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:129]: typo: no comma needed in "…destructive actions _like Delete_ or Remove"

File: docs/pages/web/iconbuttonfloating.js [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+      </MainSection>
+      <AccessibilitySection name={generatedDocGen?.displayName}>
+        <MainSection.Subsection
+          title="ARIA attributes"
+          description={`
+IconButtonFloating conveys the component behavior using iconography. IconButton requires \`accessibilityLabel\`, a text description for screen readers to announce and communicate the represented [Icon](/web/icon). In the example below, the screen reader reads: "Create Pin menu". **The label should describe the intent of the action, not the Icon itself.** For example, use "Edit board" instead of "Pencil".
+
+If IconButtonFloating is used as a control button to show/hide a Popover-based component, we recommend passing the following ARIA attributes to assist screen readers:
+
+- \`accessibilityControls\`: informs the screen reader that IconButton controls the display of an anchored Popover-based component. It populates [aria-controls](https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html).
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:153]: these references to IconButton need to be updated

File: docs/pages/web/iconbuttonfloating.js [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+          description={`
+          **Floating**
+          IconButtonFloating is lifted off of the background with shadows built-in ($elevation-floating). The floating elevation serves as an affordance for floating actions.
+
+          In dark mode, the elevation is achieved with colors instead of shadows ($color-background-elevation-floating). See [elevation guidelines](/foundations/elevation) for reference. `}
+        />
+        <MainSection.Subsection
+          title="Placement"
+          description={`
+          A consistent position improves discoverability as it appears across a responsive range of surfaces. In most cases, only one IconButtonFloating should be present on a screen. The exception is using a centered IconButtonFloating as a primary action, like board creation.
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:197]: grammar nit: it's not clear what "it" refers to in the first sentence

File: docs/pages/web/iconbuttonfloating.js [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+
+          **Default**
+          IconButtonFloating default placement is set to stick to the bottom-end of the screen. This applies to supportive actions, such as opening related content and resources.
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:200]: typo: "_IconButtonFloating's_ default placement…"

File: docs/pages/web/iconbuttonfloating.js [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+
+          **Bottom-center**
+          Use bottom-center placement when leading Pinners to an action or task, such as creating a new board.
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:203]: This is a little awkward — I think we either need to call out `bottom-center` as a prop value (using backticks) or use an article ("Use _the_ bottom-center placement…")

File: docs/pages/web/iconbuttonfloating.js [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+`}
+        />
+      </MainSection>
+      <MainSection name="Writing">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            description={`
+            - Use a descriptive label to describe the IconButton action by beginning with a verb.`}
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:214]: IconButtonFloating

File: docs/pages/web/iconbuttonfloating.js [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+      </MainSection>
+      <QualityChecklist component={generatedDocGen?.displayName} />
+      <MainSection name="Related">
+        <MainSection.Subsection
+          description={`
+**[Button](/web/button)**
+Button allows users to take actions, and make choices using text labels to express what action will occur when the user interacts with it.
+
+**[Icon](/web/icon)**
+IconButtons use icons instead of text to convey available actions on a screen. Use an existing one from the Gestalt [Icon](/web/icon) library.
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:231]: This doesn't seem to refer to Icon?

File: packages/gestalt/src/IconButtonFloating.js [hunk 16]
```diff
@@ -0,0 +1,90 @@
+// @flow strict
+import { forwardRef, type Node } from 'react';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+import icons from './icons/index.js';
+
+type Props = {|
+  /**
+   * Specifies the `id` of an associated element (or elements) whose contents or visibility are controlled by IconButton so that screen reader users can identify the relationship between elements. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
```
Comment (dangerismycat) [inline packages/gestalt/src/IconButtonFloating.js:9]: IconButtonFloating

File: packages/gestalt/src/IconButtonFloating.js [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+  /**
+   * Toggles between binary states: on/off, selected/unselected, open/closed
+   */
+  selected?: boolean,
+|};
+
+type unionRefs = HTMLButtonElement | HTMLAnchorElement;
+
+/**
+ * A [IconButtonFloating](https://gestalt.pinterest.systems/web/iconbuttonfloating) represents the primary or most common action on the screen. As the name suggests, it floats over the content and is always on top of everything on the screen. Similar to [IconButton](https://gestalt.pinterest.systems/web/iconbutton), the floating version uses icons instead of text to convey available actions; however, it is used when the action needs to be visible at all times in a sticky way where content can scroll underneath. IconButtonFloating remains in place on scroll.
```
Comment (dangerismycat) [inline packages/gestalt/src/IconButtonFloating.js:48]: No article before the initial "IconButtonFloating"

File: docs/pages/web/iconbuttonfloating.js (hunk 13 above)
Comment (dangerismycat) [reply docs/pages/web/iconbuttonfloating.js:203]: Oh wait — now I'm confused. This section reads as if we controlled placement and these are two props the user can select from. But this is just design guidance? The current format is a bit confusing to me.

Comment (ayeshakmaz) [reply docs/pages/web/iconbuttonfloating.js]: Updated this, let me know if it sounds better

File: docs/pages/web/iconbuttonfloating.js [hunk 18]
```diff
@@ -0,0 +1,248 @@
...
+        <MainSection.Subsection
+          title="Placement"
+          description={`
+          IconButtonFloating is always placed along the bottom of the screen. A consistent position improves discoverability when IconButtonFloating appears across a responsive range of surfaces. For example, an IconButtonFloating used to open resources should remain in the same spot on the page across surfaces. In most cases, only one IconButtonFloating should be present on a screen. The exception is using a centered IconButtonFloating as a primary action, like board creation.
+
+          **Bottom edge placement**
+          IconButtonFloating should typically be placed in the bottom edge of the screen (bottom right for LTR languages, and bottom left for RTL languages). This applies to supportive actions, such as opening related content and resources.
+
+          **Centered placement**
+          Use centered placement when leading Pinners to an action or task that should remain present when scrolling, such as creating a new board.
```
Comment (dangerismycat) [inline docs/pages/web/iconbuttonfloating.js:203]: grammar nit: "Use _a_ centered placement…"

File: packages/gestalt/src/IconButtonFloating.js (hunk 7 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/IconButtonFloating.js]: I'm confused about the comment (color) relating to the my comment (compression). Seems unrelated?

File: packages/gestalt/src/IconButtonFloating.js [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+   */
+  accessibilityLabel: string,
+  /**
+   * Icon displayed in IconButtonFloating to convey the behavior of the component. Refer to the [iconography](/foundations/iconography/library) guidelines regarding the available icon options.
+   */
+  icon: $Keys<typeof icons>,
+  /**
+   * Callback fired when the component is clicked, pressed or tapped.
+   */
+  onClick?: ({|
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:31]: I think onClick should be requires as the main action for IconButton is to "open" a Modal and or Dropdown, onClick must be used to trigger that response.

File: packages/gestalt/src/IconButtonFloating.js [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+
+type Props = {|
+  /**
+   * Specifies the `id` of an associated element (or elements) whose contents or visibility are controlled by IconButtonFLoating so that screen reader users can identify the relationship between elements. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityControls?: string,
+  /**
+   * Indicates that IconButtonFloating hides or exposes collapsible components and expose whether they are currently expanded or collapsed. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityExpanded?: boolean,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:15]: accessibilityExpanded should only required with Dropdowns right, not necessarily Modals? 

If so, then ok if optional. We should explain this in Docs.

File: packages/gestalt/src/IconButtonFloating.js [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+import { forwardRef, type Node } from 'react';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+import icons from './icons/index.js';
+
+type Props = {|
+  /**
+   * Specifies the `id` of an associated element (or elements) whose contents or visibility are controlled by IconButtonFLoating so that screen reader users can identify the relationship between elements. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityControls?: string,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:11]: accessibilityControls should only required with Dropdowns right, not necessarily Modals?

If so, then ok if optional. We should explain this in Docs. 

If not, if it controls Modal too, then it should be required.

File: packages/gestalt/src/IconButtonFloating.js [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+  /**
+   * Indicates that IconButtonFloating hides or exposes collapsible components and expose whether they are currently expanded or collapsed. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityExpanded?: boolean,
+  /**
+   * Indicates that this component controls the appearance of interactive popup elements, such as Dropdown or Modal. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityHaspopup?: boolean,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:19]: why is this a prop, and we don't just set it to true?

It should be a default right? I would remove this prop.

File: packages/gestalt/src/IconButtonFloating.js [hunk 23]
```diff
@@ -0,0 +1,90 @@
...
+  onClick?: ({|
+    event:
+      | SyntheticMouseEvent<HTMLButtonElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  |}) => void,
+  /**
+   * Toggles between binary states: on/off, selected/unselected, open/closed
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:40]: I think the description here should get scoped to the Floating objective: MOdal/Dropdown open/closed

File: packages/gestalt/src/IconButtonFloating.js [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+   */
+  selected?: boolean,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:42]: I think it should be also required right?

Not 100% sure if required to be selected with Modals, but for sure with Dropdown. If valid for both cases, I would make it required.

File: packages/gestalt/src/IconButtonFloating.js [hunk 25]
```diff
@@ -0,0 +1,90 @@
...
+    accessibilityLabel,
+    icon,
+    onClick,
+    selected,
+  }: Props,
+  ref,
+): Node {
+  return (
+    <Box borderStyle="shadow" rounding="circle">
+      <IconButton
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:71]: Are this iconButtons meant to have a Tooltip as well? The future of IconButtons is to all of the have tooltips? Thoughts?

@ponori

File: packages/gestalt/src/IconButtonFloating.js [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+    accessibilityHaspopup,
+    accessibilityLabel,
+    icon,
+    onClick,
+    selected,
+  }: Props,
+  ref,
+): Node {
+  return (
+    <Box borderStyle="shadow" rounding="circle">
```
Comment (AlbertCarreras) [inline packages/gestalt/src/IconButtonFloating.js:70]: More thoughts here: https://www.loom.com/share/4edf15fe0a324fadb6f1ef25d9ad8989

File: docs/pages/web/iconbuttonfloating.js [hunk 27]
```diff
@@ -0,0 +1,248 @@
...
+            - Only if it is the most suitable way to present a screen's high-emphasis action.
+        `}
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            title="When not to use"
+            description={`
+            - There isn't a need for a fixed IconButtonFloating visible all the time on the screen.
+            - To replace [IconButton](/web/iconbutton) established patterns, such as nav elements.
```
Comment (AlbertCarreras) [inline docs/pages/web/iconbuttonfloating.js:102]: I'd do navigation rather than "nav"

File: docs/pages/web/iconbuttonfloating.js [hunk 28]
```diff
@@ -0,0 +1,248 @@
...
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            description="Use the designated size and style."
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Scale or style IconButtonFloating. Consistent button sizes promote a cohesive user experience."
```
Comment (AlbertCarreras) [inline docs/pages/web/iconbuttonfloating.js:141]: This is more intended for designers... As an eng you can't modify tis component's style

File: docs/pages/web/iconbuttonfloating.js [hunk 29]
```diff
@@ -0,0 +1,248 @@
...
+If IconButtonFloating is used as a control button to show/hide a Popover-based component, we recommend passing the following ARIA attributes to assist screen readers:
+
+- \`accessibilityControls\`: informs the screen reader that IconButtonFloating controls the display of an anchored Popover-based component. It populates [aria-controls](https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html).
+- \`accessibilityHaspopup\`: informs the screen reader that there’s a Popover-based component attached to IconButtonFloating. It populates [aria-haspopup](https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html).
+- \`accessibilityExpanded\`: informs the screen reader whether an anchored Popover-based component is currently open or closed. It populates [aria-expanded](https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html).
+`}
+        >
+          <MainSection.Card cardSize="lg" />
+          <MainSection.Subsection
+            title="Keyboard interaction"
```
Comment (AlbertCarreras) [inline docs/pages/web/iconbuttonfloating.js:160]: Is this "Keyboard interaction" or "Landmark" or "Positioning".

Keyboard interaction seems a little bit less explicit. I see the connection but I see some disconnect title<>content

File: docs/pages/web/iconbuttonfloating.js [hunk 30]
```diff
@@ -0,0 +1,248 @@
...
+          />
+          <MainSection.Card cardSize="lg" />
+        </MainSection.Subsection>
+      </AccessibilitySection>
+      <MainSection name="Localization" description="Be sure to localize `accessibilityLabel`." />
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Size"
+          columns={2}
+          description="IconButtonFloating size is only available in a single size, 56px. Keeping the size consistent will promote a cohesive Pinner experience and avoid usability issues. "
```
Comment (AlbertCarreras) [inline docs/pages/web/iconbuttonfloating.js:172]: nit 

maybe 

"xl" (56px)

File: docs/pages/web/iconbuttonfloating.js [hunk 31]
```diff
@@ -0,0 +1,248 @@
...
+        />
+        <MainSection.Subsection
+          title="Color"
```
Comment (AlbertCarreras) [inline docs/pages/web/iconbuttonfloating.js:175]: Why do we need this section? It's just a darkMode section?

File: docs/pages/web/iconbuttonfloating.js [hunk 32]
```diff
@@ -0,0 +1,248 @@
...
+
+          - Dark Gray Icon ("darkGray") + White ("white") background + $elevation-floating
+          Color combination in light-mode UI.
+
+          -  White ("white") Icon + $color-background-elevation-floating
+          Color combination in dark-mode UI.
+          `}
+        />
+        <MainSection.Subsection
+          title="Elevation"
```
Comment (AlbertCarreras) [inline docs/pages/web/iconbuttonfloating.js:187]: This seems very implementation-y to me. 

I don't think we need a lot of "foundations" content here. 

I think i'd be great to explain how to elevate the component: position=absolute or fixed, Some might attempt Layer!
I'd explain the implementation for elevation

**Review by AlbertCarreras — APPROVED** [review]
It works as is, left comments across the files. I'm feel somehow strong about some of them but I'll let you decide which ones makes sense to implent.

File: packages/gestalt/src/IconButtonFloating.js (hunk 22 above)
Comment (ayeshakmaz) [reply packages/gestalt/src/IconButtonFloating.js:19]: haspopup doesn't apply for modals, only menus

File: packages/gestalt/src/IconButtonFloating.js (hunk 24 above)
Comment (ayeshakmaz) [reply packages/gestalt/src/IconButtonFloating.js:42]: selected would only apply for Dropdowns

**Comment by ayeshakmaz** [comment]
@AlbertCarreras made updates to the props, had to add a new one. Expanded and Controls only apply for Dropdowns, so can't make them required

Removed Color section

Tried wrapping in TapArea - it still gives the halo issue, and then also compresses the button too much (since scale transforms compound). So for now I think leaving as-is is best, and then if there are any issues while building out examples, I'll make adjustments

File: docs/docs-components/COMPONENT_DATA.js [hunk 33]
```diff
@@ -1170,6 +1170,56 @@ const GENERAL_COMPONENT_LIST: $ReadOnlyArray<ListItemType> = [
...
+    category: 'Actions',
+    status: {
+      accessible: {
+        summary: 'ready',
+        a11yVisual: 'ready',
+        a11yScreenreader: 'ready',
+        a11yNavigation: 'ready',
+        a11yComprehension: 'ready',
+      },
+      badge: null,
```
Comment (AlbertCarreras) [inline docs/docs-components/COMPONENT_DATA.js:1188]: NEW?

**Pull request merged**